src/hipscat/pixel_tree/moc_filter.py

```diff
@@ -0,0 +1,67 @@
+import numba
+import numpy as np
+from mocpy import MOC
+from numba import njit
+
+from hipscat.pixel_tree.pixel_tree import PixelTree
+
+
+def filter_by_moc(
+    tree: PixelTree,
+    moc: MOC,
+) -> PixelTree:
+    """Filters a pixel tree to only include the pixels that overlap with the pixels in the moc
+
+    Args:
+        tree (PixelTree): The tree to perform the filtering on
+        moc (mocpy.MOC): The moc to use to filter
+
+    Returns:
+        A new PixelTree object with only the pixels from the input tree that overlap with the moc.
+    """
+    moc_ranges = moc.to_depth29_ranges
+    tree_29_ranges = tree.tree << (2 * (29 - tree.tree_order))
+    tree_mask = perform_filter_by_moc(tree_29_ranges, moc_ranges)
+    return PixelTree(tree.tree[tree_mask], tree.tree_order)
+
+
+@njit(
+    numba.bool_[::1](
+        numba.int64[:, :],
+        numba.uint64[:, :],
+    )
+)
+def perform_filter_by_moc(
```

Comment:
Please add more test coverage here. This is finnicky stuff!

Reply:
It is actually all covered by the tests that go over all the cases, it's the problem that numba compiled code doesn't show up in the coverage. But I can add a comment so that codecov doesn't count it